migration-guide/drl.adoc

```diff
@@ -0,0 +1,376 @@
+    }
+
+    public MeasurementUnit(DataStore<Measurement> measurements) {
+        this.measurements = measurements;
+    }
+
+    public DataStore<Measurement> getMeasurements() {
+        return measurements;
+    }
+
+    public Set<String> getControlSet() {
+        return controlSet;
+    }
+}
+----
+
+This Rule Unit class associates rules (DRL) and inserted facts (DataSource), so it helps you to manage the relationship.
+In addition to the API, the DRL syntax style also changes from traditional DRL, as in the following example:
+
+.Traditional DRL
+[source,drl]
+----
+global java.util.Set controlSet;
+
+rule "will execute per each Measurement having ID color"
+when
+	Measurement( id == "color", $colorVal : val )
+then
+	controlSet.add($colorVal);
+end
+----
+
+to Rule Unit DRL, as in the following example:
+
+.Rule Unit DRL
+[source,drl]
+----
+unit MeasurementUnit;
+
+rule "will execute per each Measurement having ID color"
+when
+	/measurements[ id == "color", $colorVal : val ]
+then
+	controlSet.add($colorVal);
+end
+----
+
+Firstly, declare `unit`, don't need to declare `global`, because `controlSet` is a field of `MeasurementUnit`.
+Traditional global usage would become the Unit's field.
+In addition, the main difference is `OOPath notation` in left hand side (LHS) of the rule.
+It is described in detail at Rule conditions in DRL.
+You should change from the class name (Measurement) to the DataSource property name in the Unit class (measurements).
+
+In addition, you need to take care of the insert in the right hand side (RHS) of the rule.
+The `insert` method inserts a fact into a DEFAULT entry-point, but in Rule Unit use cases, you must specify a DataSource where you want to insert the fact.
+Hence, you must call `DataStore.add` or `DataStream.append` instead of insert.
+
+.Traditional DRL
+[source,drl]
+----
+rule "Insert another Measurement"
+when
+	Measurement( id == "color", $colorVal : val )
+then
+	insert(new Measurement("monochrome", "false"));
+end
+----
+
+to Rule Unit DRL, as in the following example:
+
+.Rule Unit DRL
+[source,drl]
+----
+unit MeasurementUnit;
+
+rule "Insert another Measurement"
+when
+	/measurements[ id == "color", $colorVal : val ]
+then
+	measurements.add(new Measurement("monochrome", "false"));
+end
+----
+
+[NOTE]
+====
+You don't need to re-write update, modify and delete, because they work for a fact in its DataSource expectedly.
+These changes are not negligible, but hopefully, as this migration guide demonstrated, they are not so difficult to implement either.
+====
+
+== Migration from non-executable model to executable model
+
+The `drools-engine-classic` and `drools-mvel` dependencies execute rules with **non executable model**.
+`drools-engine` executes rules with **executable model**.
+As mentioned, `drools-engine-classic` and `drools-mvel` are deprecated, it is recommended to migrate from a non-executable model to an executable model by changing dependencies.
```

Comment:
in the scope of product, this is up to PM... but just to be clear:

MVEL and Classic Engine are first class citizen in the community, they are not deprecated at all.

Executable model and things like Rule Units are more modern ways to work with Drools, but it doesn't mean that others are not.

Reply:
Need to check the community docs as much of this content came from that, but I have adjusted based on your feedback regarding what's deprecated and what's not.